exporters/trace/jaeger/env_test.go

```diff
@@ -286,6 +86,72 @@ func TestNewRawExporterWithEnvImplicitly(t *testing.T) {
 	assert.Equal(t, password, uploader.password)
 }
 
+func TestAgentEndpointFromEnv(t *testing.T) {
+	testCases := []struct {
+		name             string
+		envAgentHost     string
+		envAgentPort     string
+		host             string
+		port             string
+		expectedHostPort string
+	}{
+		{
+			name:             "overrides HostPort value via environment variables",
+			envAgentHost:     "localhost",
+			envAgentPort:     "6832",
+			host:             "hostNameToBeReplaced",
+			port:             "8203",
+			expectedHostPort: "localhost:6832",
+		},
+		{
+			name:             "envAgentHost is empty, will not overwrite host value",
+			envAgentHost:     "",
+			envAgentPort:     "6832",
+			host:             "hostNameNotToBeReplaced",
+			port:             "8203",
+			expectedHostPort: "hostNameNotToBeReplaced:6832",
+		},
+		{
+			name:             "envAgentPort is empty, will not overwrite port value",
+			envAgentHost:     "localhost",
+			envAgentPort:     "",
+			host:             "hostNameToBeReplaced",
+			port:             "8203",
+			expectedHostPort: "localhost:8203",
+		},
+		{
+			name:             "envAgentHost and envAgentPort are empty, will not overwrite HostPort value",
+			envAgentHost:     "",
+			envAgentPort:     "",
+			host:             "hostNameNotToBeReplaced",
+			port:             "8203",
+			expectedHostPort: "hostNameNotToBeReplaced:8203",
+		},
+	}
+
+	envStore := ottest.NewEnvStore()
+	envStore.Record(envAgentHost)
+	envStore.Record(envAgentPort)
+	defer func() {
+		require.NoError(t, envStore.Restore())
+	}()
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			require.NoError(t, os.Setenv(envAgentHost, tc.envAgentHost))
+			require.NoError(t, os.Setenv(envAgentPort, tc.envAgentPort))
+			host, port := agentEndpointFromEnv()
+			if host == "" {
+				host = tc.host
+			}
+			if port == "" {
+				port = tc.port
+			}
+			hostPort := fmt.Sprintf("%s:%s", host, port)
+			assert.Equal(t, hostPort, tc.expectedHostPort)
```

Comment:
nit: the call order here is expected then actual

```suggestion
			assert.Equal(t, tc.expectedHostPort, hostPort)
```